AshleyLab/motrpac-frontend
# MOTR-50 Auth0 Authentication Integration

**Key Changes:**
1. Created a new **Auth0** account for the project with the **Tenant** named `motrpac-project`.
2. Integrated the **Auth0** authentication service into the `motrpac-frontend` codebase.
3. Updated various unit tests due to changes as part of the **Auth0** implementation.
4. Added temporary placeholder pages for the `^/methods`, '^/team`, and `^/contact` pages.
5. Added **Coming Soon!** mouseover UIs to the sub-analysis pages.
6. Temporarily disabled the "Download Items" button on `^/download` page for demo purpose.
7. Updated various Storybook behaviors due to Auth0 implementation.

***

**Technical Notes:**
1. New users should sign up with **Auth0** using **Google Auth** when prompted to login.
2. `user_metadata` should be added to users' profiles in **Auth0**.

***

**Steps to test:**
1. Click on the **Submitter Login** button in the footer and expect to be redirected to **Auth0**'s _Universal Login_ page.
2. Sign up and/or login, and expect the following:
    - to be redirected to the `^/dashboard` page if authenticated.
    - to see the logged-in user's gravatar and first name being shown in the footer.
    - to see the extra nav items (e.g. _Dashboard_, _Analysis_, _Methods_, _Data_) being shown in the header.
3. Click on the **[first/given name] Logout** button in the footer and expect to be redirected to the landing page. Also expect to see the nav items described above no longer being shown.
4. Login again and go to the `^/analysis/animal` or `^/analysis/human` pages.
5. Click on the **Published Data Meta - Analysis** panel to navigate to the _Animal/Human Data Analysis_ pages. Mouseover either one of the 2 panels in the UI and expect to see the **Coming Soon!** overlay being shown.
6. Click on either one of the **Methods**, **Who We Are** or **Contact** nav items in the header. Expect to see **Coming Soon!** being shown on the pages.
